net/api/impl/posix_server.h

```diff
@@ -126,24 +137,29 @@ class HTTPServerPOSIX final {
   void Thread(Socket socket) {
     // TODO(dkorolev): Benchmark QPS.
     while (!terminating_) {
-      HTTPServerConnection connection(socket.Accept());
-      if (terminating_) {
-        break;
-      }
-      const HTTPReceivedMessage& message = connection.Message();
-      const std::string& path = message.Path();
-      const url::URL url(path);
-      Request request{url, connection, message};
-      {
-        // TODO(dkorolev): Read-write lock for performance?
-        std::lock_guard<std::mutex> lock(mutex_);
-        const auto cit = handlers_.find(url.path);
-        if (cit != handlers_.end()) {
-          // TODO(dkorolev): Properl handle the shutdown case when the handler spawns another thread.
-          cit->second(std::move(request));
-        } else {
-          connection.SendHTTPResponse("", HTTPResponseCode::NotFound);
+      try {
+        HTTPServerConnection connection(socket.Accept());
+        if (terminating_) {
+          break;
+        }
+        const HTTPReceivedMessage& message = connection.Message();
+        const std::string& path = message.Path();
+        const url::URL url(path);
+        Request request{url, connection, message};
+        {
+          // TODO(dkorolev): Read-write lock for performance?
+          std::lock_guard<std::mutex> lock(mutex_);
+          const auto cit = handlers_.find(url.path);
+          if (cit != handlers_.end()) {
+            // TODO(dkorolev): Properly handle the shutdown case when the handler spawns another thread.
+            cit->second(std::move(request));
+          } else {
+            connection.SendHTTPResponse("", HTTPResponseCode::NotFound);
```

Comment:
Fixed in 8e590db.

(The logic is going to change to indicate that a particular handler is now active, so that it's not deregistered while running. Stay tuned.)